### PR TITLE
add maven publishing configuration

### PR DESCRIPTION
**Pull request by mirceanis**
This PR adds the configuration needed for publishing the sdk libraries to a maven repo.

This fixes [#162411452](https://www.pivotaltracker.com/story/show/162411452) from pivotal as well.
Developers can click through SDK sources right from android studio.

### CODE REVIEW

File: publishing.gradle [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                        url siteUrl
+
+                        licenses {
+                            license {
+                                name "The Apache Software License, Version 2.0"
+                                url "http://www.apache.org/licenses/LICENSE-2.0.txt"
+                            }
+                        }
+                        developers {
+                            developer {
```
Comment (mirceanis) [inline publishing.gradle]: I only added myself because this requires an email address.

Comment (ugoamanoh) [reply publishing.gradle]: ok

File: publishing.gradle [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+        Properties publishProps = new Properties()
+        propFile.withInputStream {
+            publishProps.load(it)
+        }
+        project.ext.publishProps = publishProps
+    } else {
+        project.ext.publishProps = null
+    }
+}
```
Comment (ugoamanoh) [inline publishing.gradle]: @mirceanis Is there a way to test this script?

**Comment by ugoamanoh** [comment]
@mirceanis LGTM

File: publishing.gradle (hunk 2 above)
Comment (mirceanis) [reply publishing.gradle]: Yes, this exact script was used to publish v0.3.2 to jCenter as well

**Pull request merged**